.github/workflows/code_scanning.yaml

```diff
@@ -0,0 +1,54 @@
+# Copyright 2024 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: "CodeQL"
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+    branches:
+      - main
+      - release-*
+  push:
+    branches:
+      - main
+      - release-*
+  schedule:
+    - cron: '31 11 * * 4'
```

Comment:
Question: What is the benefit of a `cron` job?

Reply:
Catch a CVE or a new malpractice

Reply:
For projects such as the device plugin where we're already triggering this on PRs -- including dependency bumps -- and on merge, I don't think this is required. Let's rather remove this and add scheduled scans as a follow-up once we are sure that the existing signals are not sufficient.

Reply:
Done